packages/core/src/StateNode.ts

```diff
@@ -405,60 +407,18 @@ export class StateNode<
     return selectedTransition ? [selectedTransition] : undefined;
   }
 
-  public get initialStateValue(): StateValue | undefined {
-    if (this.__cache.initialStateValue) {
-      return this.__cache.initialStateValue;
-    }
-
-    let initialStateValue: StateValue | undefined;
-
-    if (this.type === 'parallel') {
-      initialStateValue = mapFilterValues(
-        this.states as Record<string, StateNode<TContext, any, TEvent>>,
-        state => state.initialStateValue || EMPTY_OBJECT,
-        stateNode => !(stateNode.type === 'history')
-      );
-    } else if (this.initial !== undefined) {
-      if (!this.states[this.initial]) {
-        throw new Error(
-          `Initial state '${this.initial}' not found on '${this.key}'`
-        );
-      }
-
-      initialStateValue = (isLeafNode(this.states[this.initial])
-        ? this.initial
-        : {
-            [this.initial]: this.states[this.initial].initialStateValue
-          }) as StateValue;
-    }
-
-    this.__cache.initialStateValue = initialStateValue;
-
-    return this.__cache.initialStateValue;
-  }
-
   /**
    * The target state value of the history state node, if it exists. This represents the
    * default state value to transition to if no history value exists yet.
    */
-  public get target(): StateValue | undefined {
-    let target;
+  public get target(): string | undefined {
```

Comment:
this probably could be removed in the future, once we represent history state as a state with default transition etc, but for now, I've decided to keep it to reduce the scope of the PR